### PR TITLE
Make sure that IdFieldType#isAggregatable is accurate.

### LINKED ISSUE #62897: field _id is not aggregatable but field_caps says its aggregatable
original kibana issue - https://github.com/elastic/kibana/issues/77206

Related to - https://github.com/elastic/elasticsearch/pull/49166

### PR DESCRIPTION
**Pull request by jtibshirani**
Before, it always returned 'true' even when the setting
"indices.id_field_data.enabled" was false.

Fixes #62897.

### CODE REVIEW

**Comment by jtibshirani** [comment]
@elasticmachine run elasticsearch-ci/2

File: server/src/main/java/org/elasticsearch/index/mapper/IdFieldMapper.java [hunk 1]
```diff
@@ -156,11 +165,6 @@ public Query termsQuery(List<?> values, QueryShardContext context) {
                     CircuitBreakerService breakerService,
                     MapperService mapperService
                 ) {
-                    if (mapperService.isIdFieldDataEnabled() == false) {
-                        throw new IllegalArgumentException("Fielddata access on the _id field is disallowed, "
-                            + "you can re-enable it by updating the dynamic cluster setting: "
-                            + IndicesService.INDICES_ID_FIELD_DATA_ENABLED_SETTING.getKey());
-                    }
                     deprecationLogger.deprecate("id_field_data", ID_FIELD_DATA_DEPRECATION_MESSAGE);
```
Comment (jtibshirani) [inline server/src/main/java/org/elasticsearch/index/mapper/IdFieldMapper.java:168]: I kept the deprecation message here so we don't issue a warning on every call to field caps.

**Review by jimczi — APPROVED** [review]
LGTM

**Comment by jtibshirani** [comment]
The refactor #63197 was merged concurrently, and happens to fix the bug. So after merging with master, this PR is now a test-only change that verifies the fix.

**Comment by jtibshirani** [comment]
@elasticmachine run elasticsearch-ci/1
@elasticmachine run elasticsearch-ci/2

**Pull request merged**